### ISSUE TITLE
Could you please elaborate the following line? why not use the y coming from batch?

### ISSUE THREAD

**Issue by nooralahzadeh** [oscarknagg/few-shot#10]
https://github.com/oscarknagg/few-shot/blob/672de83a853cc2d5e9fe304dc100b4a735c10c15/experiments/maml.py#L98

**Comment by nooralahzadeh** [comment]
I think I figured it out. You consider the set of classes whose contribute in the n-shot task. However the 'y' coming from batch has all the classes. am I right?

**Comment by oscarknagg** [comment]
Yes thats right. The y coming from the batch is the underlying class from the whole dataset (>1000 classes for Omniglot, 80 for miniImageNet). We don't want want to use the true class just the label of each sample within that particular n-shot task. Nicely figured out!

**Comment by lyrgwlr** [comment]
> Yes thats right. The y coming from the batch is the underlying class from the whole dataset (>1000 classes for Omniglot, 80 for miniImageNet). We don't want want to use the true class just the label of each sample within that particular n-shot task. Nicely figured out!

But the predict of the model('logits') seems would be optimized like:
[[1, 0, 0, 0, 0], [0, 1, 0, 0, 0], [0, 0, 1, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]] because of the fixed label [0, 1, 2, 3, 4]. I'm so confused about that. Could you please explain this? @nooralahzadeh @oscarknagg , thanks a lot!!

**Comment by TomStarshak** [comment]
I think I agree with @lyrgwlr here. Why can't the model learn to ignore the information and just output the correct answer depending on the position of the task?
